starknet-rpc-test/add_deploy_account_transaction.rs

```diff
@@ -0,0 +1,99 @@
+extern crate starknet_rpc_test;
+
+use std::vec;
+
+use rstest::rstest;
+use starknet_accounts::AccountFactory;
+use starknet_core::types::{BlockId, BlockTag, DeployAccountTransactionResult};
+use starknet_ff::FieldElement;
+use starknet_providers::Provider;
+use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, CAIRO_1_ACCOUNT_CONTRACT_CLASS_HASH, SIGNER_PRIVATE};
+use starknet_rpc_test::fixtures::madara;
+use starknet_rpc_test::utils::{build_deploy_account_tx, build_oz_account_factory, create_account, AccountActions};
+use starknet_rpc_test::{MadaraClient, Transaction, TransactionResult};
+
+#[rstest]
+#[tokio::test]
+async fn fail_execution_step_with_no_storage_change(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    // deploy account
+    let oz_factory =
+        build_oz_account_factory(rpc, "0x123", FieldElement::from_hex_be(CAIRO_1_ACCOUNT_CONTRACT_CLASS_HASH).unwrap())
```

Comment:
Isn't "0x123" also an invalid private key. Could it fail because of that rather than the lack of fees?

Reply:
My question is does it fail because of lack of fee, or because of an invalid signature before this?

Reply:
But maybe you are right and `0x123` is a valid signature. But earlier we have this tests
```rust
async fn fail_validation_step(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
    let madara = madara.await;
    let rpc = madara.get_starknet_client();

    // using incorrect private key to generate the wrong signature
    let account = create_account(rpc, "0x1234", ARGENT_CONTRACT_ADDRESS, true);
``` 

Which make me think you signature is not valid in this test too, and we fail before the fee thing, just because of bad signature

Reply:
Right. Actually "0x123" is misleading here, I get what you meant now. So in the context of the Argent Account, 0x2 will be an incorrect key but since DeployAccount doesn't have any validation (it happens during the contract run I beleive?), entering any private key here will pass the validation step. I guess I will use the original private key to avoid this confusion.

Reply:
I think you can use whatever, but just mention why it is ok in a comment :)

Reply:
I have fixed this but it's in the new PR, sorry!